src/components/seo.tsx

```diff
@@ -103,6 +120,23 @@ const SEO: React.FC<SeoProps> = ({
         type="font/woff2"
         crossOrigin="anonymous"
       />
+      {origin &&
+        pathname &&
+        LANGUAGES.map((lang) => {
+          const href = `${origin}${lang === 'en' ? '' : `/${lang}`}`;
+          pathname =
+            language !== lang && translation
+              ? prependAndAppendTrailingSlash(translation)
+              : pathname;
+          return (
+            <link
+              rel="alternate"
+              href={href.concat(pathname)}
+              hrefLang={lang}
+              key={lang}
+            />
```

Comment:
so this seems to add an alternate entry to refer to the translated version. Imagine you replace that entire code with a function named `listLocalizedVersions()` like so

```suggestion
      {listLocalizedVersions()}
```

That would make it so much more readable because I know what the piece of code is intended to do. Inside the function implementation I could then add more information regarding the alternate links because this is not common knowledge. I looked it up here https://developers.google.com/search/docs/advanced/crawling/localized-versions and only then I understood what's happening here. This cost me at least 5min to understand. You can help your future self by separating the code into a separate file, give the function a name and provide some comments about the implementation.

This tip is valid for most of the lengthy files I have seen in this PR, maybe you could invest some refactoring. Either within this PR or afterwards (to come to an end because the PR got too big already).